src/uct/ib/base/ib_device.c

```diff
@@ -176,6 +195,76 @@ static void uct_ib_device_get_locality(const char *dev_name,
     *numa_node = (status == UCS_OK) ? n : -1;
 }
 
+static void
+uct_ib_device_async_event_dispatch(uct_ib_device_t *dev,
+                                   const uct_ib_async_event_t *event)
+{
+    uct_ib_async_event_val_t *entry;
+    khiter_t iter;
+    int ret;
+
+    ucs_spin_lock(&dev->async_event_lock);
+    iter  = kh_put(uct_ib_async_event, &dev->async_events_hash, *event, &ret);
```

Comment:
can use get now, looks like no need for put

Reply:
that was the purpose of the flag - it just looks like no need.
we do nothing if it's new entry, but we still add it so following put will return KEY_PRESENT